File: app/src/main/java/com/example/matt/concoctioncrafter/data/RecipeDAO.java
/* Copyright Matthew Coughlin 2018 */

package com.example.matt.concoctioncrafter.data;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface RecipeDAO {
    @Query("SELECT * FROM `recipes.db`")
    List<Recipe> getAll();

    @Query("SELECT * FROM `recipes.db` ORDER BY recipe_name ASC")
    LiveData<List<Recipe>> getAllRecipes();

    @Query("SELECT * FROM `recipes.db` WHERE id IN (:recipeIds)")
    List<Recipe> loadAllByIds(int[] recipeIds);

    @Query("SELECT * FROM `recipes.db` WHERE id = :id LIMIT 1")
    Recipe findById(int id);

    @Query("SELECT * FROM `recipes.db` WHERE recipe_name = :name LIMIT 1")
    Recipe findByName(String name);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Recipe recipe);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Recipe... recipes);

    @Update
    void update(Recipe recipe);

    @Query("DELETE FROM `recipes.db`")
    void deleteAll();

    @Delete
    void delete(Recipe recipe);
}

File: app/src/main/java/com/example/matt/concoctioncrafter/data/RecipeDAO.kt
/* Copyright Matthew Coughlin 2018, 2019 */

package com.example.matt.concoctioncrafter.data

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface RecipeDAO {
    @get:Query("SELECT * FROM `recipes.db`")
    val all: List<Recipe>

    @get:Query("SELECT * FROM `recipes.db` ORDER BY _recipeName ASC")
    val allRecipes: LiveData<List<Recipe>>

    @Query("SELECT * FROM `recipes.db` WHERE _recipeName = :name LIMIT 1")
    fun findByName(name: String): Recipe

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(recipe: Recipe)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(vararg recipes: Recipe)

    @Update
    fun update(recipe: Recipe)

    @Query("DELETE FROM `recipes.db`")
    fun deleteAll()

    @Delete
    fun delete(recipe: Recipe)
}
